Modules/faulthandler.c

```diff
@@ -1373,6 +1378,15 @@ _PyFaulthandler_Init(int enable)
        signal handler uses more than SIGSTKSZ bytes of stack memory on some
        platforms. */
     stack.ss_size = SIGSTKSZ * 2;
+#ifdef AT_MINSIGSTKSZ
+    /* bpo-46968: Query Linux for minimal stack size to ensure signal delivery
+       for the hardware running CPython. This OS feature is available in
+       Linux kernel version >= 5.14 */
+    unsigned long at_minstack_size = getauxval(AT_MINSIGSTKSZ);
+    if (at_minstack_size != 0) {
+        stack.ss_size = SIGSTKSZ + at_minstack_size;
```

Comment:
I'm surprised that AT_MINSIGSTKSZ doesn't return directly the value which should be passed to sigaltstack(), but it seems like your code is the correct way to use the value.

Reply:
The `AT_MINSIGSTKSZ` is a preprocessor variable. The `getauxval` checks hardware capabilities (e.g. total size of CPU state to be saved at context switches) and may return different values on different platforms.

Reply:
I mean that I expected to write (simplified code without error handling): ``stack.ss_size = getauxval(AT_MINSIGSTKSZ)``, rather than ``stack.ss_size = SIGSTKSZ + getauxval(AT_MINSIGSTKSZ)``. But your code looks correct according to the doc that I found on the Internet. Like the official Linux documentation:

```
ss.ss_size = getauxval(AT_MINSIGSTKSZ) + SIGSTKSZ;
```

ref: https://docs.kernel.org/x86/elf_auxvec.html#introduction